sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpFileCacheSensor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SonarC#
+ * Copyright (C) 2014-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.csharp;
+
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonarsource.dotnet.shared.plugins.AbstractFileCacheSensor;
+import org.sonarsource.dotnet.shared.plugins.HashProvider;
+
+public class CSharpFileCacheSensor extends AbstractFileCacheSensor {
+  public CSharpFileCacheSensor(CSharp cSharp, HashProvider hashProvider) {
+    super(cSharp, hashProvider);
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor.name("C# file caching sensor");
+    descriptor.onlyOnLanguage(CSharpPlugin.LANGUAGE_KEY);
+  }
```

Comment:
Can the base class describe itself to avoid duplication? It knows the language so it should have name and key to do it